src/tcluser.c

```diff
@@ -350,27 +350,79 @@ static int tcl_addbot STDVAR
     } else {
       q = strchr(argv[2], '/');
     }
+
+    /* Verify ports */
+    if (q) {
+      /* Split and count */
+      *q++ = 0;
+      p = strchr(q, '/');
+      if (p)
+        *p++ = 0;
+
+#ifndef TLS
+      /* Check TLS */
+      if ((q && *q == '+') || (p && *p == '+')) {
+        Tcl_AppendResult(irp, "0", NULL);
+        return TCL_OK;
+      }
+
+#endif
+      /* Verify */
+      if (!check_int_range(q, 0, 65536)) {
+        Tcl_AppendResult(irp, "0", NULL);
+        return TCL_OK;
+      }
+      /* check_int_range returns 0 if p is NULL */
+      if (p && !check_int_range(p, 0, 65536)) {
+        Tcl_AppendResult(irp, "0", NULL);
+        return TCL_OK;
+      }
+
+    }
+
     userlist = adduser(userlist, argv[1], "none", "-", USER_BOT);
     bi = user_malloc(sizeof(struct bot_addr));
 #ifdef TLS
     bi->ssl = 0;
 #endif
+
+    /* Max addr len is 60 ? (see cmd_pls_bot in cmds.c) */
+    if ((count = strlen(argv[2])) > 60) {
+      count = 60;
+      argv[2][count] = 0;
+    }
+    /* Trim IPv6 []s out if present */
+    if (braced) {
+      --count;
+      argv[2][count] = 0;
+      memmove(argv[2], argv[2] + 1, count);
```

Comment:
modifying argv[n] content for the tcl string procs can break things (it should be treated as const)